modules/@apostrophecms/rich-text-widget/ui/apos/tiptap-extensions/Default.js

```diff
@@ -0,0 +1,64 @@
+// If a style has been marked `def`, we need to make sure it
+// and it's attributes are prioritized in our editor.
+// It's easier to create a new Node with our specifications and put it
+// at the front of the line than try to infer between editor instantiation
+// and the editor-user setting styles via the toolbar to know when its needed.
```

Comment:
This marks a different approach to the whole problem, a few things led up to this:
- We allow multiple styles that use the same tag (and different attributes/options). When one of these multiples was marked `def` we couldn't infer whether it was time to use the settings of our default or if the element settings were being applied explicitly from the toolbar (switching among styles)
- In Prosemirror all heading levels are part of the same Node extension, making it difficult to single out a specific level and apply attributes to, esp when compounded by the multiple styles/same tag problem in point one^

So instead of trying to neatly configure all our nodes we create a new node where we can reliably stamp all our requirements in, knowing that when we prioritize the this new node type in the document's schema, it will be the default element on first creation.

This node is a one-off, in that there are no controls to set any future selections to this type. The editor essentially hands it off to be transformed by the normal style controls, where it is converted among the normal base nodes/marks.